express-notes-api/index.js

```diff
@@ -0,0 +1,212 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const inputFileName = 'data.json'
+
+// Clients can GET a list of notes.
+
+app.get('/api/notes', (req, res) => {
+
+    fs.readFile(inputFileName, 'utf8', (err, data) => {
+        if (err) {
+            res.status(500);
+            res.json({"error": "Error reading data file"})
+        } else {
+            try {
+                const content = JSON.parse(data);
+                const outArray = [];
+                for (const outObj in content.notes) {
+                    outArray.push({
+                        "content": content.notes[outObj].content,
+                        "id": content.notes[outObj].id
+                    })
+                }
+                res.status(200)
+                res.json(outArray);
+            } catch (err) {
+                res.status(500);
+                res.json({"error": "Error parsing JSON string"})
+            }
+        } 
+    });
+})
+
+
+// Clients can GET a single note by id.
+
+
+app.get('/api/notes/:id', (req, res) => {
+    if (parseInt(req.params.id) <= 0 || typeof parseInt(req.params.id) !== 'number' || parseInt(req.params.id) % 1 !== 0) {

```

Comment:
`parseInt` drops the fractional part of `req.params.id`, so you're not actually testing if `id` is an integer:

---

![image](https://user-images.githubusercontent.com/16143771/130552313-c5e6b920-c6c8-47c8-b58f-cdf8ef92b2d5.png)